epics/ca.py

```diff
@@ -1035,10 +1035,12 @@ def array_cast(data, count, ntype, use_numpy):
             out = memcopy(data)
         return out
 
-    def unpack(data, count, ntype, use_numpy):
+    def unpack(data, count, ntype, use_numpy, elem_count):
         "simple, native data type"
         if data is None:
             return None
+        elif ntype == dbr.CHAR and elem_count > 1:
```

Comment:
Original comment was:
> Is count == 1 in that case?
> If count > 1, it's an array for sure. If count==1 it might be an array.

As a reminder: `elem_count` is the total element count of the channel, not necessarily the size of the current value held within. `count` is the client-requested number of elements. When the client requests 0 (or `None`) elements, `count` gets automatically set to the size of the current value (which is `<= elem_count`).

With that, I believe that the `elem_count` as reported by EPICS is the key here, not the `count`. If the EPICS layer says that this as a char array and not a single char value (i.e., `elem_count > 1`), we should always get a char array back.